OpenRA.Mods.Common/Traits/Buildings/RallyPoint.cs

```diff
@@ -73,22 +75,32 @@ public IEnumerable<IOrderTargeter> Orders
 		public Order IssueOrder(Actor self, IOrderTargeter order, Target target, bool queued)
 		{
 			if (order.OrderID == "SetRallyPoint")
-				return new Order(order.OrderID, self, false) { TargetLocation = self.World.Map.CellContaining(target.CenterPosition), SuppressVisualFeedback = true };
+				return new Order(order.OrderID, self, false) { TargetLocation = self.World.Map.CellContaining(target.CenterPosition), SuppressVisualFeedback = true,
+					ExtraData = ((RallyPointOrderTargeter)order).SetPrimaryBuilding ? ExtraDataSetPrimaryBuilding : 0 };
 
 			return null;
 		}
 
 		public void ResolveOrder(Actor self, Order order)
 		{
 			if (order.OrderString == "SetRallyPoint")
+			{
 				Location = order.TargetLocation;
+				if (order.ExtraData == ExtraDataSetPrimaryBuilding)
+				{
+					var primaryBuildingTrait = self.TraitOrDefault<PrimaryBuilding>();
```

Comment:
I would prefer it if we didn't have RallyPoint talk to PrimaryBuilding – the former is a general RTS feature, while the latter is specific to the weird production model of the early C&Cs (and isn't used in our TD mod at all).

A clean way around this would be to rename `ExtraDataSetPrimaryBuilding` to `ForceSet` or similar, and then have `PrimaryBuilding` check for the `SetRallyPoint` order string and ExtraData itself.
You can avoid this by renaming `ExtraDataSetPrimaryBuilding` to